### ISSUE TITLE
AI Assistant with Google Gemini interrupts while streaming output

### ISSUE THREAD

**Issue by notpeter** [zed-industries/zed#17515]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Steps to reproduce:
1. Configure Google Gemini in the AI Assistant
2. Have a prompt that generates some text. Longer is better.
3. Watch tokens stream in and then get interrupted.

Unrelated, Google Gemini (internal-only) is also broken via zed.dev provider (screenshot).

### Environment

Zed: v0.153.0 (Zed Nightly 938c90fd3bd2bba920d35f112aca445f293a46d8)
OS: macOS 14.6.1
Memory: 64 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="637" alt="Screenshot 2024-09-06 at 14 58 01" src="https://github.com/user-attachments/assets/622bb9dd-b354-4356-8757-d7657138a229">

zed.dev error:
<img width="391" alt="Screenshot 2024-09-06 at 14 59 58" src="https://github.com/user-attachments/assets/1012dfc8-cda9-4897-b12d-3af8400cb487">

### If applicable, attach your Zed.log file to this issue.

```
2024-09-06T14:55:25.720881-04:00 [WARN] request completed with error: unknown error
2024-09-06T14:55:26.128829-04:00 [ERROR] error during countTokens, status code: 400, body: {
  "error": {
    "code": 400,
    "message": "CountTokens requires generate_content_request or contents to be set.",
    "status": "INVALID_ARGUMENT"
  }
}
2024-09-06T14:55:35.206449-04:00 [WARN] request completed with error: unknown error
2024-09-06T14:55:42.259709-04:00 [WARN] request completed with error: unknown error
2024-09-06T14:55:48.139879-04:00 [WARN] request completed with error: unknown error
```

**Comment by saroshwahla** [comment]
Error still persists on Zed 0.156.0 (same Environment as noted above).

Interestingly, identical successive queries in quick succession will yield an error in the first or second attempt but the third attempt is successful. Maybe because the results are cached?

**Comment by notpeter** [comment]
This is fixed in main and will available in Zed 0.158.x which is this week's Zed Preview and next week's Zed Stable.
